packages/salesforce/src/Utils.js

```diff
@@ -134,3 +134,69 @@ export async function pollJobResult(conn, job, pollInterval, pollTimeout) {
     attempt++;
   }
 }
+
+/**
+ * Flattens an array of nested objects to a maximum of two levels deep.
+ * - Each object in the array is flattened to have dot-separated keys.
+ * - Throws an error if the input is not an array or if a nested object exceeds two levels deep.
+ *
+ * @param {Array<Object>} input - An array of objects to flatten.
+ * @returns {Array<Object>} - An array of flattened objects.
+ * @throws {Error} - Throws an error if the input is not an array of objects or if nesting exceeds two levels.
+ */
+
+export function flattenData(input) {
```

Comment:
Open question - should this be `flatten` or `removeNestings`?

When you talk about flattening an array, I think about taking an array of arrays and "flattening" them into a single array of elements.

But what we're doing here is removing the nesting of an object. 

The inverse operation would be to take an array of objects and nest them - detect any dot referenced properties and convert them into a nested object, broken on the dot. Talking about `nesting` or `validateNoNesting` is easier to understand than `validateNoDotKeys`

And generally, we  can talk about nesting objects, un-nesting objects (a bit awkward bit it works), or removing nesting of objects.

Phew :sweat_smile: 

Having written this out I'm pretty sure we should rename this to `removeNestings`

Reply:
Yeah it's hard naming function, i think `removeNestings` is the best fit for this use case although the keyword `remove` sounds like we are completely removing the keys from the object 🤷🏽

Reply:
I have rename the function to `removeNestings`